### DISCUSSION TITLE
NO DELAY Option

### DISCUSSION THREAD

**Discussion by ghost** [rathole-org/rathole#244] [Q&A]
Do i need to set NO DELAY per service and per server transport? The debug output is confusing to me 
![image](https://user-images.githubusercontent.com/47906826/236346082-5dea2f01-6b1c-4fb6-8342-3302138b0427.png)
DEBUG OUTPUT

SERVER-TOML
![Untitled](https://user-images.githubusercontent.com/47906826/236346781-63d127d7-197c-4d07-a0bd-c5ec25a95091.png)

So do i only need to place NODELAY once per server? or once per SERVER and CLIENT?

**Comment by yujqiao** [comment]
You need to set nodelay on both server and client.

Per-service configuration should work. What you report suggests a bug 🤔. It could be something wrong calculating the service config given the global config.

**Reply by ghost** [reply]
Currently I only have the server setup, early next week I will have my server here and will configure the client with NO DELAY and report back.

By the way love the project, rathole always works seamlessly! Appreciate the quick response!
